operator/pkg/reconciliation/construct_podtemplatespec_test.go

```diff
@@ -410,6 +410,280 @@ func TestCassandraDatacenter_buildPodTemplateSpec_initcontainers_merge(t *testin
+			},
+		},
+	}
+
+	podTemplateSpec, err := buildPodTemplateSpec(dc, map[string]string{zoneLabel: "testzone"}, "testrack")
+
+	assert.NoError(t, err, "should not have gotten error when building podTemplateSpec")
+
+	initContainers := podTemplateSpec.Spec.InitContainers
+
+	assert.Equal(t, 1, len(initContainers))
+	assert.Equal(t, ServerConfigContainerName, initContainers[0].Name)
+
+	serverConfigInitContainer := initContainers[0]
+	assert.Equal(t, 1, len(serverConfigInitContainer.VolumeMounts))
+	// We use a contains check here because the ordering is not important
+	assert.True(t, volumeMountsContains(serverConfigInitContainer.VolumeMounts, volumeMountNameMatcher("server-config")))
+
+	volumes := podTemplateSpec.Spec.Volumes
+	assert.Equal(t, 4, len(volumes))
+	// We use a contains check here because the ordering is not important
+	assert.True(t, volumesContains(volumes, volumeNameMatcher("server-config")))
+	assert.True(t, volumesContains(volumes, volumeNameMatcher("test-data")))
+	assert.True(t, volumesContains(volumes, volumeNameMatcher("server-logs")))
+	assert.True(t, volumesContains(volumes, volumeNameMatcher("encryption-cred-storage")))
+
+	containers := podTemplateSpec.Spec.Containers
+	assert.Equal(t, 3, len(containers))
+
+	testContainer := findContainer(containers, "test")
+	assert.NotNil(t, testContainer)
+
+	assert.Equal(t, 3, len(testContainer.VolumeMounts))
+	// We use a contains check here because the ordering is not important
+	assert.True(t, volumeMountsContains(testContainer.VolumeMounts, volumeMountNameMatcher(PvcName)))
+	assert.True(t, volumeMountsContains(testContainer.VolumeMounts, volumeMountNameMatcher("test-data")))
+	assert.True(t, volumeMountsContains(testContainer.VolumeMounts, volumeMountNameMatcher("server-config")))
+
+	cassandraContainer := findContainer(containers, CassandraContainerName)
+	assert.NotNil(t, cassandraContainer)
+
+	cassandraVolumeMounts := cassandraContainer.VolumeMounts
+	assert.Equal(t, 4, len(cassandraVolumeMounts))
+	assert.True(t, volumeMountsContains(cassandraVolumeMounts, volumeMountNameMatcher("server-config")))
+	assert.True(t, volumeMountsContains(cassandraVolumeMounts, volumeMountNameMatcher("server-logs")))
+	assert.True(t, volumeMountsContains(cassandraVolumeMounts, volumeMountNameMatcher("encryption-cred-storage")))
+	assert.True(t, volumeMountsContains(cassandraVolumeMounts, volumeMountNameMatcher("server-data")))
+
+	loggerContainer := findContainer(containers, SystemLoggerContainerName)
+	assert.NotNil(t, loggerContainer)
+
+	loggerVolumeMounts := loggerContainer.VolumeMounts
+	assert.Equal(t, 1, len(loggerVolumeMounts))
+	assert.True(t, volumeMountsContains(loggerVolumeMounts, volumeMountNameMatcher("server-logs")))
+}
+
+type VolumeMountMatcher func(volumeMount corev1.VolumeMount) bool
+
+type VolumeMatcher func(volume corev1.Volume) bool
+
+func volumeNameMatcher(name string) VolumeMatcher {
+	return func(volume corev1.Volume) bool {
+		return volume.Name == name
+	}
+}
```

Comment:
I'm good with this, but I feel obliged to point out Rob Pike's repo here -> https://github.com/robpike/filter

Reply:
Do you want me to update the code to use that? Happy to do so. I promise you I do not have NIH syndrome :)

Reply:
to get rid of all the one line functions and functions abstracting a for loop? no it reads cleanly here, and this is test code, if it's working right now let's keep it as-is